.github/workflows/hackage.yml

```diff
@@ -0,0 +1,127 @@
+name: Hackage
+
+on:
+  push:
+    branches:
+      - '*-hackage'
+
+jobs:
+  check-and-upload-tarballs:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package: ["hie-compat", "hls-graph", "shake-bench",
+                  "hls-plugin-api", "ghcide", "hls-test-utils",
+                  "hls-brittany-plugin", "hls-floskell-plugin", "hls-fourmolu-plugin",
+                  "hls-ourmolu-plugin", "hls-stylish-haskell-plugin",
+                  "hls-class-plugin", "hls-eval-plugin", "hls-explicit-imports-plugin",
+                  "hls-haddock-comments-plugin", "hls-hlint-plugin",
+                  "hls-module-name-plugin", "hls-pragmas-plugin",
+                  "hls-refine-imports-plugin", "hls-retrie-plugin",
+                  "hls-splice-plugin", "hls-tactics-plugin",
+                  "hls-call-hierarchy-plugin",
+                  "haskell-language-server"]
+        ghc: ["8.10.6", "8.8.4", "8.6.5"]
+
+    steps:
+
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - uses: haskell/actions/setup@v1
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-version: '3.4'
+
+      - name: Cache Cabal
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-cabal
+        with:
+          path: |
+            ~/.cabal/packages
+            ~/.cabal/store
+          key: v2-${{ runner.os }}-${{ matrix.ghc }}-hackage-${{ hashFiles('**/*.cabal') }}
```

Comment:
i think we could reuse the test/bench cache here

Reply:
The total cache limit is still 5GB so I would say reusing the test cache would be much preferred to avoid over-running the limit:

https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy